chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
I propose using Module.clock and Module.reset since they don't actually collide with anything.  It makes more sense that the implicit clock is always named "clock", rather than sometimes "clock" and other times "implicitClock".